packages/js/client-config-builder/README.md

```diff
@@ -42,15 +42,470 @@ const config = builder.build();
 let client = new PolywrapClient(config);
 ```
 
-## Methods
+## Types
```

Comment:
Above this line, in the introduction section, I think it could use some work. A few thoughts:
- I don't think "DSL" is properly used here.
- Ideally all code snippets should be from-source so that they remain up-to-date, so this one that's not a snippet worries me as it can become out-of-date.
- I think should start with introducing the main concepts a user must understand to get started. We can introduce the intended usage pattern of `init -> configure -> build`, and we can point to sub-sections for each. "Initialization" shows the constructor signature. "Configure" shows the different "add..." methods. etc
- Lastly, in this intro section, having an e2e example snippet (taken from a buildable source file) would be great to have.

So in summary:
- Brief Description
- Intro Concepts (linking to reference sections further down)
- E2E Example Snippet

Apologies if this is overly verbose :P

Reply:
i reworked this intro and used snippets as suggested in your other comment